openedx/openedx-app-android
# feat: Firebase config

This PR  returns the default Firebase configuration flow with the `google-services.json` file.
The manual configuration will be implemented for the Learning site project in a separate module.

1. **Removed** manual configuration from the `OpenEdXApp.kt`
```
val options = FirebaseOptions.Builder()
                .setProjectId(firebaseConfig.projectId)
                .setApplicationId(firebaseConfig.applicationId)
                .setApiKey(firebaseConfig.apiKey)
                .setGcmSenderId(firebaseConfig.gcmSenderId)
                .build()
Firebase.initialize(this, options)
```

2. **Introduced** the `generateGoogleServicesJson` function in `ConfigHelper.groovy`.
```
def generateGoogleServicesJson(applicationId) {
        def config = fetchConfig()
        def firebase = config.get("FIREBASE")
        if (!firebase) {
            return
        }
        if (!firebase.getOrDefault("ENABLED", false)) {
            return
        }

        def googleServicesJsonPath = projectDir.path + "/app/"
...
```

3. The field `GCM_SENDER_ID` was **replaced** with `PROJECT_NUMBER` in the `FirebaseConfig.kt` file.

**How to test:**
1. Create a new Firebase project
2. Configure the `config.yaml` file with the necessary data
3. Launch the app and make a test crash
4. Check the crash data in the Firebase crashlytics console

Notes:
google-services.json generated by Firebase
```
{
  "project_info": {
    "project_number": "5322XXXXX9180",
    "project_id": "testconfigprojectXXX-XXXX",
    "storage_bucket": "testconfigprojectXXXX-XXXX.appspot.com"
  },
  "client": [
    {
      "client_info": {
        "mobilesdk_app_id": "1:532239259180:android:ec6ee6XXXXXdede113e389",
        "android_client_info": {
          "package_name": "com.XXXXX.android.app"
        }
      },
      "oauth_client": [],
      "api_key": [
        {
          "current_key": "AIzaSyByTC1ZXXXXXXEzImslozg_iO5_zPfVs8"
        }
      ],
      "services": {
        "appinvite_service": {
          "other_platform_oauth_client": []
        }
      }
    }
  ],
  "configuration_version": "1"
}
```
google-services.json generated by ConfigHelper
```
{
    "project_info": {
        "project_number": "5322XXXXX9180",
        "project_id": "testconfigprojectXXX-XXXX",
        "storage_bucket": "testconfigprojectXXXX-XXXX.appspot.com"
    },
    "client": [
        {
            "client_info": {
                "mobilesdk_app_id": "1:532239259180:android:ec6ee6XXXXXdede113e389",
                "android_client_info": {
                    "package_name": "org.openedx.app"
                }
            },
            "oauth_client": [
                
            ],
            "api_key": [
                {
                    "current_key": "AIzaSyByTC1ZXXXXXXEzImslozg_iO5_zPfVs8"
                }
            ],
            "services": {
                "appinvite_service": {
                    "other_platform_oauth_client": [
                        
                    ]
                }
            }
        }
    ],
    "configuration_version": "1"
}
```
